lib/graphql/null_non_null_error.rb

```diff
@@ -0,0 +1,22 @@
+module GraphQL
+  # Raised automatically when a field's resolve function returns `nil`
+  # for a non-null field.
+  class NullNonNullError < RuntimeError
```

Comment:
I don't find NullNonNullError to be a very readable error message.  Perhaps `InvalidNullError` would be less confusing.